test/ExecutionFramework.cpp

```diff
@@ -170,6 +174,32 @@ void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256
 		m_transactionSuccessful = (m_gas != m_gasUsed);
 }
 
+void ExecutionFramework::waitForTransaction(std::string const& _txHash) const
+{
+	for (int polls = 0; true; polls++)
+	{
+		std::this_thread::sleep_for(std::chrono::milliseconds(1));
+		auto pendingBlock = m_rpc.eth_getBlockByNumber("pending", false);
+
+		bool isEmpty = pendingBlock["transactions"].empty();
+
```

Comment:
I think @chriseth's idea was to just have
```
if (!pendingBlock["transactions"].empty())
{
  BOOST_REQUIRE_EQUAL(pendingBlock["transactions"][0].asString(), _txHash);
  return;
}
```

Then you only need the ``if (polls == 200)`` case below inside the ``for`` and can just always error out after the ``for``.

Reply:
(given that you change ``true`` in the for condition to ``polls < 3000``)